File: tasks/task2/booking-service/src/main/java/com/hotelio/bookingservice/entity/BookingEvent.java
package com.hotelio.bookingservice.entity;

import java.time.Instant;

/**
 * Kafka event emitted when a booking is created.
 * Serialized as JSON via Spring Kafka's JsonSerializer.
 */
public record BookingEvent(
        Long id,
        String userId,
        String hotelId,
        String promoCode,
        Double discountPercent,
        Double price,
        Instant createdAt
) {

    public static BookingEvent from(Booking booking) {
        return new BookingEvent(
                booking.getId(),
                booking.getUserId(),
                booking.getHotelId(),
                booking.getPromoCode(),
                booking.getDiscountPercent(),
                booking.getPrice(),
                booking.getCreatedAt()
        );
    }
}

File: tasks/task2/booking-history-service/DTO/BookingEvent.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace BookingHistoryService.DTO;

public record BookingEvent
{
    [Key]
    [JsonPropertyName("id")]
    public long? Id { get; set; } = null!;

    [JsonPropertyName("userId")]
    public string UserId { get; set; } = null!;

    [JsonPropertyName("hotelId")]
    public string HotelId { get; set; } = null!;

    [JsonPropertyName("promoCode")]
    public string PromoCode { get; set; } = null!;

    [JsonPropertyName("discountPercent")]
    public double? DiscountPercent { get; set; } = null!;

    [JsonPropertyName("price")]
    public double? Price { get; set; } = null!;

    [JsonPropertyName("createdAt")]
    public double? CreatedAt { get; set; } = null!;
}